### ISSUE TITLE
Payment Gateway or Streaming Payment on top of BNBChain

### ISSUE THREAD

**Issue by huihzhao** [bnb-chain/community-contributions#52]
## Main Objectives/Goals:
The ​​objective is to build an all-in-one payment solution based on BNBChain’s tech stack, bringing more utilities to stablecoins and BNB within the BNBChain ecosystem. This integrated solution will enhance the functionality and value of digital assets, providing seamless and efficient payment options for users and fostering broader adoption and utility within the ecosystem
## Challenge Description:
Crypto payments have gained some traction within Web3 and Web2, but their adoption has not met expectations. BNBChain now offers gasless solutions and excels in performance on opBNB, which will significantly lower barriers to mass adoption. With the adoption of BEP336 on opBNB, gas fees are expected to decrease by over 90%, creating a strong synergy with payment solutions.

BNBChain also provides a comprehensive solution to build a dedicated payment Layer 2 (L2), further optimizing user experiences. Moreover, stablecoins on BNBChain are enthusiastic about collaborating closely with these payment platforms. Whether you're a payment gateway connecting merchants or a streaming payment platform supporting payroll, in-game transactions, or other functionalities, BNBChain can meet your requirements. The community eagerly anticipates these advancements in adoption.

**Comment by 9012Jessie** [comment]
Several payment gateways already support BNB Chain, such as [NowPayments](https://nowpayments.io/) and [Lunu](https://lunu.io/), with more on the way. Streaming payment is another scenario we want to support further."

**Comment by degenbuilder7** [comment]
Hi @huihzhao @9012Jessie , This sounds good, BEP336 on opBNB, gas fees are expected to decrease by over 90%, creating a strong synergy with payment solutions..

My query is regarding are we looking to focus on privacy related payments too ? or just the public ones ?

**Comment by huihzhao** [comment]
I would say no preference 🙂

**Comment by huihzhao** [comment]
Yes, the cost and performance barrier have been removed by BEP336 and performance optimisation of opbnb. Payment scenario, especially the stable coin payments can be benefited and it is also affordable to sponsor end users 😄

**Comment by huihzhao** [comment]
Now payment - https://nowpayments.io/
NOWPayments is a payment gateway that allows everyone to accept crypto payments on their websites, online stores, and social media accounts. It’s a non-custodial service, meaning it doesn’t hold or store your funds in any way. NOWPayments supports more than 50 cryptocurrencies, and it offers low transaction fees.
How does it works？

It's a crypto payment gateway with 300+ supported cryptocurrencies and a very friendly UX. Now payment can help customers organize and track the payments, monitor and track the status of payments. 

The integration flow will be like:
Sign up and set up account
Choose the payment tools
- API
- Invoice
- Plugin
- Etc
Accept payment and monitor your profit
- Fiat to crypto
- Crypto to crypto
- etc
https://nowpayments.io/help/about-nowpayments
What makes Now payment best-suited to execute this Mission?
Now Payment has supported BEP20 token for payment, e.g.USDT, USDC, FDUSD, BNB,  and has integrated with lots of merchants, e.g. Travala to explore the huge opportunities online and offline.

**Comment by huihzhao** [comment]
Lunu - https://lunu.io/
Lunu provides the cryptocurrency payment processing. They enable merchants to accept numerous cryptocurrencies as a payment method with the ability to convert the earnings to fiat currencies like Euros or U.S. Dollars.

Lunu has the following products: merchant payment processing for businesses offline via Lunu Terminals, online via Lunu Widget and Lunu Wallet for the customers to buy, store, swap, pay with crypto and withdraw to fiats.


How does it works？
With Lunu, customers can use cryptocurrencies for everyday purchases both in-store and online. The customer pays in cryptocurrency - the retailer gets paid in local currency except the Lunu fee.

Lunu has the following products: 
- Merchant payment processing for businesses offline via Lunu Terminals, 
- Online via Lunu Widget 
- Lunu Wallet for the customers to buy, store, swap, 
- Pay with crypto by link
- withdraw to fiats.


![Image](https://github.com/user-attachments/assets/2d70c8e5-2a9e-40d7-a828-87ee2bfe11be)




https://lunu.io/products#processing-service
What makes Lunu best-suited to execute this Mission?
Lunu provides comprehensive approaches for customers to proceed the crypto payment, and bridge the gap between merchants and end Web3 residents. Lunu has partnered with Shopify, WooCommerce, etc to amplify the crypto payment journey.  Lunu will onboard more users to BNB Chain ecosystem, simplify users’ crypto payment journey, and expand the use scenarios in Web2 world.

**Comment by codeytechie** [comment]
Hi @huihzhao ,

**Project Name: Payroll Protocol**

**Overview**:  
[Payroll Protocol](https://payroll-bnb.vercel.app/) is a confidential on-chain money distribution platform built on the BNB. The platform ensures the confidentiality, security, and efficiency of fund distribution by leveraging Fully Homomorphic Encryption (FHE) cryptography and smart contracts. Integrating Biconomy's AA SDK, BNB, and INCO FHEVM, Payroll Protocol enhances user experience while keeping all transaction details secure and confidential. You can watch a detailed overview of the project [here](https://youtu.be/5shVuHiOoNY).

**Key Features**:

1. **Encrypted USDC Distributions**:  
   - Employs advanced encryption techniques to protect sensitive payroll information.
   
2. **User-Friendly Interface**:  
   - Simplifies payroll processing with a single click, utilizing Biconomy's SDK for seamless interactions.
   
3. **Stealth Hold**:  
   - Allows users to hold stablecoins without revealing amounts, maintaining financial privacy.
   
4. **Underline Distribution**:  
   - Hides withdrawals by distributing encrypted amounts to multiple encrypted addresses, ensuring privacy similar to Tornado Cash.

**Architecture & Flow Diagrams**:

- The architecture leverages **INCO FHEVM** to provide hidden states for storing encrypted addresses and amounts.
- **Hyperlane infrastructure** is modified to securely bridge state values between BNB and INCO.
- Flow diagrams visually represent the confidential distribution process, demonstrating how the platform handles encrypted transactions.

**Tech Stack**:

- **Frontend**:  
   - **Next.js**, **Tailwind CSS** for building a responsive and intuitive user interface.
   
- **Authentication & Wallets**:  
   - **Privy** for secure user authentication and wallet management.
   
- **Backend & Smart Contracts**:  
   - **Node.js** and **Solidity** for backend logic and smart contract development.
   
- **Blockchain**:  
   - **BNB** for on-chain operations and **INCO FHEVM** for managing encrypted states.
   
- **Account Abstraction & Gasless Transactions**:  
   - **Biconomy** to enable seamless user interactions without the need for gas fees.
   
- **Encryption**:  
   - FHE schemes to ensure all transaction details remain confidential.

**Comment by LevanIlashvili** [comment]
@huihzhao  I’m gonna work on this in the scope of BNB hack Bangkok
